ui/analyse/src/wiki.ts

```diff
@@ -5,15 +5,33 @@ export type WikiTheory = (nodes: Tree.Node[]) => void;
 export default function wikiTheory(): WikiTheory {
   const cache = new Map<string, string>();
   const show = (html: string) => {
-    $('.analyse__wiki').html(html).toggleClass('empty', !html);
+    $('.analyse__wiki').toggleClass('empty', !html);
+    $('.analyse__wiki-text').html(html);
     site.pubsub.emit('chat.resize');
   };
 
+  $(window).on('load', () => {
+    const wikiBookField = $('#wikibook-field').first();
+    const wikiBookStorageKey = 'analyse.wikibooks.display';
+    const toggleWikiBookStorage = () => site.storage.boolean(wikiBookStorageKey).toggle();
+
+    if (!site.storage.boolean(wikiBookStorageKey).getOrDefault(true)) {
+      wikiBookField.addClass('form-fieldset--toggle-off');
+    } else {
+      site.storage.boolean(wikiBookStorageKey).set(true);
+    }
+
+    $(wikiBookField)
+      .children('legend')
+      .on('click', () => toggleWikiBookStorage())
+      .on('keypress', e => e.key == 'Enter' && toggleWikiBookStorage());
+  });
+
   const plyPrefix = (node: Tree.Node) =>
     `${Math.floor((node.ply + 1) / 2)}${node.ply % 2 === 1 ? '._' : '...'}`;
 
   const wikiBooksUrl = 'https://en.wikibooks.org';
-  const apiArgs = 'redirects&origin=*&action=query&prop=extracts&formatversion=2&format=json&exchars=1200';
+  const apiArgs = 'redirects&origin=*&action=query&prop=extracts&formatversion=2&format=json&exchars=1000';
```

Comment:
As with the implementation of the fieldset, some width was decreased for the text content, I kept the character length 1000 not to have a scrollbar for most cases, though after implementation I also felt, the scrollbar could rather be kept within the fieldset component and be much better. Please let me know your thoughts to confirm the suitability.

Reply:
Yes the scroll should be inside the fieldset. And the margin around the fieldset removed.

Reply:
Updated, thanks.